### PR TITLE
No longer use private API function generate_action_token

### PR DESCRIPTION
**Pull request by iionly**

### CODE REVIEW

File: views/default/event_calendar/personal_toggle_js.php [hunk 1]
```diff
@@ -2,17 +2,13 @@
...
 ?>
 
 <script type="text/javascript">
 function event_calendar_personal_toggle(event_id, user_id) {
 
-	var link = "<?php echo elgg_get_site_url(); ?>action/event_calendar/toggle_personal_calendar?";
-	link += "user_id="+user_id+"&event_id="+event_id+"&other=true";
-	link += "<?php echo $tokens; ?>";
+	var link = "<?php echo elgg_get_site_url(); ?>action/event_calendar/toggle_personal_calendar";
+	link += elgg.security.addToken("?user_id="+user_id+"&event_id="+event_id+"&other=true");
```
Comment (juho-jaakkola) [inline views/default/event_calendar/personal_toggle_js.php]: Even better would be to replace `$.get()` with `elgg.action()` which both:
- Formats the URL (you won't have to add wwwroot manually)
- Adds the tokens automatically

See http://learn.elgg.org/en/1.10/guides/javascript.html#ajax-helper-functions

**Comment by iionly** [comment]
Okay. Switch to elgg.action done.

**Comment by juho-jaakkola** [comment]
Could you squash the commits?

**Comment by iionly** [comment]
Sqash done.

**Pull request merged**